pkg/client/cache/reflector.go

```diff
@@ -215,3 +220,8 @@ loop:
 	glog.V(4).Infof("Watch close - %v total %v items received", r.expectedType, eventCount)
 	return nil
 }
+
+// LastSyncResourceVersion is the resource version observed when last sync with the underlying store
+func (r *Reflector) LastSyncResourceVersion() string {
+	return r.lastSyncResourceVersion
```

Comment:
There are two cases I need:

1. as a caller of reflector, I need to have a way of getting the latest known resource version
2. a way to know if what i have as last known resource version is still accurate

The IsMostRecentResourceVersion is probably needed to avoid a race condition.